python/src/gen_gherkin/generate_gherkin_feature_files_impl.py

```diff
@@ -4,43 +4,108 @@
+    Business logic for allowing gen-gherkin-behaviors command to perform Generate Gherkin feature files from AaC model behavior scenarios.
+
+    Args:
+        architecture_file (str): The YAML file containing the data models from which to generate Gherkin feature files.
+        output_directory (str): The directory into which the generated Gherkin feature files will be written.
 
     Returns:
-         The results of the execution of the gen-gherkin-behaviors command.
+        The results of the execution of the gen-gherkin-behaviors command.
     """
-
-    # TODO: implement plugin logic here
     status = ExecutionStatus.GENERAL_FAILURE
     messages: list[ExecutionMessage] = []
-    error_msg = ExecutionMessage(
-        "The gen-gherkin-behaviors command for the Generate Gherkin Feature Files plugin has not been implemented yet.",
-        MessageLevel.ERROR,
-        None,
-        None,
-    )
-    messages.append(error_msg)
 
-    return ExecutionResult(plugin_name, "gen-gherkin-behaviors", status, messages)
+    definitions_dictionary = parse(architecture_file)
+
+    results = get_template_properties(definitions_dictionary)
+
+    yaml_list = []
+    for model in results:
+        for behavior in model["behaviors"]:
+            yaml_list.append([{"behavior": behavior}])
+
+    new_file = ""
+    for yaml_object in yaml_list:
+        new_file = new_file + yaml.safe_dump_all(yaml_object, default_flow_style=False, sort_keys=False, explicit_start=True)
+
+    if len(model) < 1:
+        msg = ExecutionMessage(
+            "No applicable behavior to generate a feature file",
+            MessageLevel.ERROR,
+            None,
+            None,
+        )
+        messages.append(msg)
+        return None, ExecutionResult(plugin_name, "gen-gherkin-behaviors", ExecutionStatus.GENERAL_FAILURE, messages)
+
+    messages.append(ExecutionMessage(f"Successfully generated feature file(s) to directory: {output_directory}", MessageLevel.INFO, None, None))
+    status = ExecutionStatus.SUCCESS
+
+    return new_file, ExecutionResult(plugin_name, "gen-gherkin-behaviors", status, messages)
+
+
+def after_gen_gherkin_behaviors(architecture_file: str, output_directory: str, run_generate: Callable) -> ExecutionResult:
+    """
+    Runs Generate on the output of the gen_gherkin_behaviors plugin.
+
+    Args:
+        architecture_file (str): The YAML file containing the data models from which to generate Gherkin feature files.
+        output_directory (str): The directory into which the generated Gherkin feature files will be written.
+        run_generate (Callable): The Generation function which generates a feature file
+
+    Returns:
+        The results of the execution of the generate command.
+
+    """
+    new_file, execution_status = gen_gherkin_behaviors(architecture_file, output_directory)
+    print(new_file)
```

Comment:
This debug print statement needs to be removed.

Reply:
Oops, thought I had gotten that.